engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -79,7 +78,10 @@ def self.verify_instance(request, logger, system)
     return false unless base_product
 
     # check the cache for the system
-    cache_key = InstanceVerification.build_cache_entry(request.remote_ip, system.login, system.pubcloud_reg_code, system.proxy_byos_mode, base_product)
+    cache_key = InstanceVerification.build_cache_entry(
+      request.remote_ip, system.login, system.pubcloud_reg_code,
+      system.proxy_byos_mode, base_product, base_product.product_class
```

Comment:
In this instance of the call it is evident that the `build_cache_entry` does not need a `product_class` argument because inside of `build_cache_entry` we could just as well access the member of the `product` Object. What am I missing?

Reply:
see comment above

Reply:
I read it 3 times and I still don't understand sorry.

here we call `InstanceVerification.build_cache_entry(......, base_product, base_product.product_class)` which matches
`def self.build_cache_entry(......, product, product_class)`. Based on this signature match it appears that

`InstanceVerification.build_cache_entry(......, base_product)`
````def self.build_cache_entry(......, product)
product_class = product.product_class
```
is equivalent but does not require a signature change of the method.

Reply:
OK, if we do not use the product class as a parameter, 
[this call](https://github.com/SUSE/rmt/blob/master/engines/registry/app/models/access_scope.rb#L98-L101) does not work, as mentioned, we could query the DB for the product object instead of using the product class,
however, that would add a bunch of DB queries, so using a param instead to avoid the extra DB queries